### PR TITLE
Upgrading the repo to current android standards

### PR DESCRIPTION
**Pull request by lalwani**
The repo was not updated for several years so this diff contains a bunch of upgrades to modernize the repo. 

- Gradle 8.2.1
- Migrate to kotlin gradle dsl
- Migrating to using versions catalog instead of dependency.gradle
- AGP 8.2
- Updating build to use jdk 17 because AGP requires a gradle JVM that supports version 17
- Updating the plugin to publish to maven central https://vanniktech.github.io/gradle-maven-publish-plugin
- Using dokka 1.9
- Targeting android sdk 34
- Updated testing libs roboletric and junit
- Removed a few other deprecated plugins due to this migration
- Updated the workflow

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
         uses: actions/setup-java@v2
         with:
           distribution: 'adopt'
-          java-version: '8'
+          java-version: '17'
```
Comment (psteiger) [inline .github/workflows/build.yml:17]: I think we have some incompatibility with our current tooling + Java 17. We might need to stay on 8. @jbarr21 to confirm -- not sure if still relevant

File: core-android/src/main/java/com/uber/sdk/android/core/auth/SsoDeeplink.java [hunk 2]
```diff
@@ -31,6 +31,7 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.VisibleForTesting;
 import android.util.Log;
+
```
Comment (psteiger) [inline core-android/src/main/java/com/uber/sdk/android/core/auth/SsoDeeplink.java:34]: Why this newline? I vote to remove it

File: core-android/src/test/java/com/uber/sdk/android/core/RobolectricTestBase.java [hunk 3]
```diff
@@ -30,7 +30,7 @@
 import org.robolectric.annotation.Config;
 
 @RunWith(RobolectricTestRunner.class)
-@Config(sdk = 21)
+@Config(sdk = 26)
```
Comment (psteiger) [inline core-android/src/test/java/com/uber/sdk/android/core/RobolectricTestBase.java:33]: Might this also have unintended consequences in our CI?

File: core-android/src/test/java/com/uber/sdk/android/core/auth/LoginButtonTest.java [hunk 4]
```diff
@@ -153,7 +154,7 @@ public void testOnActivityResult_shouldCascadeLoginManager() {
 
         loginButton.onActivityResult(LoginManager.REQUEST_CODE_LOGIN_DEFAULT, 1, intent);
 
-        verify(loginManager).onActivityResult(eq(activity), eq(LoginManager.REQUEST_CODE_LOGIN_DEFAULT), eq(1),
+        verify(loginManager).onActivityResult(ArgumentMatchers.eq(activity), eq(LoginManager.REQUEST_CODE_LOGIN_DEFAULT), eq(1),
                 eq(intent));
```
Comment (psteiger) [inline core-android/src/test/java/com/uber/sdk/android/core/auth/LoginButtonTest.java:158]: Those changes here seem No-op -- and I actually find the plain `eq` more readable

File: rides-android/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        unitTests {
+            isIncludeAndroidResources = true
+        }
+
+    }
+}
+
+dependencies {
+    implementation(libs.uberRides) {
+        exclude(group = "org.slf4j", module = "slf4j-log4j12")
```
Comment (psteiger) [inline rides-android/build.gradle.kts:42]: can we add a comment on why we are excluding it?

File: rides-android/src/test/java/com/uber/sdk/android/rides/RobolectricTestBase.java [hunk 6]
```diff
@@ -30,6 +30,6 @@
  * Base test class to remove use of annotations on all test classes.
  */
 @RunWith(RobolectricTestRunner.class)
-@Config(sdk = 21)
+@Config(sdk = 26)
```
Comment (psteiger) [inline rides-android/src/test/java/com/uber/sdk/android/rides/RobolectricTestBase.java:33]: same q on unintended consequences. Will this change our internal test base default SDK ver?

File: .github/workflows/build.yml (hunk 1 above)
Comment (tyvsmith) [reply .github/workflows/build.yml:17]: We don't consume this internally, should be fine.

File: core-android/src/test/java/com/uber/sdk/android/core/RobolectricTestBase.java (hunk 3 above)
Comment (lalwani) [reply core-android/src/test/java/com/uber/sdk/android/core/RobolectricTestBase.java:33]: we don't consume this sdk internally. the min supported sdk version is 26 so should be fine.

File: rides-android/src/test/java/com/uber/sdk/android/rides/RobolectricTestBase.java (hunk 6 above)
Comment (lalwani) [reply rides-android/src/test/java/com/uber/sdk/android/rides/RobolectricTestBase.java:33]: no, this is just in the 3P repository. we don't consume it internally

**Review by psteiger — APPROVED** [review]

File: build.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        lint {
+            checkTestSources = true
+            val lintXml = file("lint.xml")
+            if (lintXml.exists()) {
+                lintConfig = lintXml
+            }
+        }
+    }
+
+    pluginManager.withPlugin("com.android.library") {
```
[the remark this answers is not in the document]
Comment (lalwani) [reply build.gradle.kts:66]: I haven't looked at how it's done in RIBs repo so not sure what you mean.
What's the benefit? I can check it later and update the gradle file if needed

**Pull request merged**